CMakeLists.txt

```diff
@@ -300,10 +300,14 @@ if(ENABLE_ASAN)
     if (NOT ENABLE_ASAN_OK)
         MESSAGE(FATAL_ERROR "The compiler does not support address sanitizer")
     endif()
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    add_compile_options(-static-libasan)
+    add_link_options(-static-libasan)
+endif()
     add_compile_options(-fsanitize=address)
     add_compile_options(-g)
     add_compile_options(-fno-omit-frame-pointer)
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
+    add_link_options(-fsanitize=address)
```

Comment:
Our requirement on CMake is 3.5.0, while `add_link_options` was introduced since 3.13, which is too restrictive for our users.